File: 2019_Qualifier_YouCanGoYourOwnWay.java
// code_report Solution
// Video Link: https://youtu.be/SIRDTt89zKk
// Problem Link: https://codingcompetitions.withgoogle.com/codejam/round/0000000000051705/00000000000881da

import java.util.*;
import java.io.*;

public class Solution {
    
    public static void solve(StringBuffer s) {
        for (int i = 0; i < s.length(); ++i) {
            char c = s.charAt(i) == 'E' ? 'S' : 'E';
            s.setCharAt(i, c);
        } 
    }

    public static void main(String []args) {
        Scanner scan = new Scanner (System.in);
        int t = scan.nextInt ();
        for (int j = 1; j <= t; j++) {
            int n = scan.nextInt ();
            StringBuffer s = new StringBuffer(scan.next());
            solve(s);
            System.out.println ("Case #" + Integer.toString(j) + ": " + s);
        }
    }
}

File: 2019_Qualifier_YouCanGoYourOwnWay.hs
-- code_report Solution
-- Video Link: https://youtu.be/SIRDTt89zKk
-- Problem Link: https://codingcompetitions.withgoogle.com/codejam/round/0000000000051705/00000000000881da

import Control.Monad
import qualified Data.ByteString.Char8 as B
readInt = readLn :: IO Int

everySecond :: [a] -> [a]
everySecond [] = []
everySecond (a:b:c) = b : everySecond c

solve :: String -> String
solve = map (\c -> if c == 'E' then 'S' else 'E')

format :: (Int, String) -> String
format (x,s) = "Case #" ++ show x ++ ": " ++ s  

main :: IO ()
main = do
    t <- readInt
    v <- replicateM (2 * t) $ getLine
    mapM_ putStrLn $ map format $ zip [1..] $ map solve $ everySecond v 
